src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -116,10 +141,58 @@ private static void GetNestedGroups(LdapConnection connection, ClaimsIdentity pr
             var group = searchResponse.Entries[0]; // Get the object that was found on ldap
             var groupDN = group.DistinguishedName;
 
+            if (processedGroups.Contains(groupDN)) {
+                // No need to continue, this group was resolved before
+                return;
+            }
+
+            retrievedClaims.Add(new KeyValuePair<string, string>(principal.RoleClaimType, groupCN));
```

Comment:
Why not add this whether or not there's any `SearchResponse.Entries` like before?

Reply:
What would be the scenario where a `groupCN` should be added to the claims despite not being found in LDAP (=> no `SearchResponse.Entries`)? If a trusted LDAP is used this should not be able to happen, but is there any reason you would like the claim being added regardless of finding the group in LDAP?